javascript/linked-list/trees/trees.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+class BTNode {
+  constructor(value, left = null, right = null) {
+    this.value = value;
+    this.left = left;
+    this.right = right;
+  }
+}
+
+class BinaryTree {
+  constructor(root = null) {
+    this.root = root;
+  }
+  preOrder(node) {
+    // could do anything but we're just logging
+    console.log(node.value);
+    // go left
+    if (node.left) {
+      this.preOrder(node.left);
+    }
+    // go right
+    if (node.right) {
+      this.preOrder(node.right);
+    }
+    return;
+  }
+}
+
+class BinarySearchTree extends BinaryTree {
+  constructor(root = null) {
+    super(root);
+  }
+  //if no root, num is now root
+  add(number, node) {
```

Comment:
Again, our initial call should only have a single argument (the value we are adding).